### ISSUE TITLE
getPurchaseData returns duplicates for Apple receipts

### ISSUE THREAD

**Issue by drinchev** [voltrue2/in-app-purchase#103]
Hey there,

I just wanted to make sure that what I've found is a bug.

When I have : 

```
in_app: 
      [ { quantity: '1',
          product_id: 'blah.blah.blah',
          transaction_id: '210000259386802',
          original_transaction_id: '210000259386802',
          purchase_date: '2016-04-14 16:03:33 Etc/GMT',
          purchase_date_ms: '1460649813982',
          purchase_date_pst: '2016-04-14 09:03:33 America/Los_Angeles',
          original_purchase_date: '2016-04-14 16:03:34 Etc/GMT',
          original_purchase_date_ms: '1460649814000',
          original_purchase_date_pst: '2016-04-14 09:03:34 America/Los_Angeles',
          expires_date: '2016-05-14 16:03:33 Etc/GMT',
          expires_date_ms: '1463241813000',
          expires_date_pst: '2016-05-14 09:03:33 America/Los_Angeles',
          web_order_line_item_id: '210000038560504',
          is_trial_period: 'false' } ] },
  latest_receipt_info: 
   [ { quantity: '1',
       product_id: 'blah.blah.blah',
       transaction_id: '210000259386802',
       original_transaction_id: '210000259386802',
       // Note Purchase date is the same as the one in in_app object
       purchase_date: '2016-04-14 16:03:33 Etc/GMT',
       purchase_date_ms: '1460649813982',
       purchase_date_pst: '2016-04-14 09:03:33 America/Los_Angeles',
       original_purchase_date: '2016-04-14 16:03:34 Etc/GMT',
       original_purchase_date_ms: '1460649814000',
       original_purchase_date_pst: '2016-04-14 09:03:34 America/Los_Angeles',
       expires_date: '2016-05-14 16:03:33 Etc/GMT',
       expires_date_ms: '1463241813982',
       expires_date_pst: '2016-05-14 09:03:33 America/Los_Angeles',
       web_order_line_item_id: '210000038560504',
       is_trial_period: 'false' },
     { quantity: '1',
       product_id: 'blah.blah.blah',
       transaction_id: '210000265773203',
       original_transaction_id: '210000259386802',
       purchase_date: '2016-05-14 16:03:33 Etc/GMT',
       purchase_date_ms: '1463241813000',
       purchase_date_pst: '2016-05-14 09:03:33 America/Los_Angeles',
       original_purchase_date: '2016-05-14 10:03:37 Etc/GMT',
       original_purchase_date_ms: '1463220217552',
       original_purchase_date_pst: '2016-05-14 03:03:37 America/Los_Angeles',
       expires_date: '2016-06-14 16:03:33 Etc/GMT',
       expires_date_ms: '1465920213000',
       expires_date_pst: '2016-06-14 09:03:33 America/Los_Angeles',
       web_order_line_item_id: '210000038560503',
       is_trial_period: 'false' } ]
```

I get a response with three purchases instead of 2.

The purchase with transaction id `210000259386802` is returned twice.

Considering the code https://github.com/voltrue2/in-app-purchase/blob/master/lib/apple.js#L226 It looks like it is supposed to be like that ( it will not duplicate only older purchases ).

I believe Apple will never give two purchases with the same `transaction_id` which differ in time. At least this is what I've noticed in my logs.

so the line should be something like : 

```
if (tids[tid] && tids[tid].time <= pdate) {
  //                                             ^ catches equals as well
  // avoid duplicate and keep the latest
  index = tids[tid].index;
}
```

Can someone else confirm that as well, so I can prepare a pull request. Thanks!

**Comment by voltrue2** [comment]
Hello

Would you attach the returned list with duplicates?

Cheers

**Comment by drinchev** [comment]
I just tried it with production data ( Removed some fields for privacy )

Receipt response after validated : 

```json
{
	"service": "apple",
	"latest_receipt": "REMOVED FOR PRIVACY",
	"latest_receipt_info": [{
		"is_trial_period": "false",
		"original_purchase_date_pst": "2017-04-11 03:10:23 America/Los_Angeles",
		"original_purchase_date_ms": "1491905423000",
		"original_purchase_date": "2017-04-11 10:10:23 Etc/GMT",
		"purchase_date_pst": "2017-04-11 03:10:23 America/Los_Angeles",
		"purchase_date_ms": "1491905423000",
		"purchase_date": "2017-04-11 10:10:23 Etc/GMT",
		"original_transaction_id": "520000222450197",
		"transaction_id": "520000222450197",
		"product_id": "de.test.product.1",
		"quantity": "1"
	}, {
		"is_trial_period": "false",
		"web_order_line_item_id": "520000040310202",
		"expires_date_pst": "2017-05-12 02:09:05 America/Los_Angeles",
		"expires_date_ms": "1494580145000",
		"expires_date": "2017-05-12 09:09:05 Etc/GMT",
		"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
		"original_purchase_date_ms": "1491988147000",
		"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
		"purchase_date_pst": "2017-04-12 02:09:05 America/Los_Angeles",
		"purchase_date_ms": "1491988145000",
		"purchase_date": "2017-04-12 09:09:05 Etc/GMT",
		"original_transaction_id": "520000222666210",
		"transaction_id": "520000222666210",
		"product_id": "db.test.product.2",
		"quantity": "1"
	}, {
		"is_trial_period": "false",
		"web_order_line_item_id": "520000040310204",
		"expires_date_pst": "2017-06-12 02:09:05 America/Los_Angeles",
		"expires_date_ms": "1497258545000",
		"expires_date": "2017-06-12 09:09:05 Etc/GMT",
		"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
		"original_purchase_date_ms": "1491988147000",
		"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
		"purchase_date_pst": "2017-05-12 02:09:05 America/Los_Angeles",
		"purchase_date_ms": "1494580145000",
		"purchase_date": "2017-05-12 09:09:05 Etc/GMT",
		"original_transaction_id": "520000222666210",
		"transaction_id": "520000229967375",
		"product_id": "db.test.product.2",
		"quantity": "1"
	}, {
		"is_trial_period": "false",
		"web_order_line_item_id": "520000042429692",
		"expires_date_pst": "2017-07-12 02:09:05 America/Los_Angeles",
		"expires_date_ms": "1499850545000",
		"expires_date": "2017-07-12 09:09:05 Etc/GMT",
		"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
		"original_purchase_date_ms": "1491988147000",
		"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
		"purchase_date_pst": "2017-06-12 02:09:05 America/Los_Angeles",
		"purchase_date_ms": "1497258545000",
		"purchase_date": "2017-06-12 09:09:05 Etc/GMT",
		"original_transaction_id": "520000222666210",
		"transaction_id": "520000237814537",
		"product_id": "db.test.product.2",
		"quantity": "1"
	}],
	"receipt": {
		"in_app": [{
			"is_trial_period": "false",
			"original_purchase_date_pst": "2017-04-11 03:10:23 America/Los_Angeles",
			"original_purchase_date_ms": "1491905423000",
			"original_purchase_date": "2017-04-11 10:10:23 Etc/GMT",
			"purchase_date_pst": "2017-04-11 03:10:23 America/Los_Angeles",
			"purchase_date_ms": "1491905423000",
			"purchase_date": "2017-04-11 10:10:23 Etc/GMT",
			"original_transaction_id": "520000222450197",
			"transaction_id": "520000222450197",
			"product_id": "db.test.product.1",
			"quantity": "1"
		}, {
			"is_trial_period": "false",
			"web_order_line_item_id": "520000040310202",
			"expires_date_pst": "2017-05-12 02:09:05 America/Los_Angeles",
			"expires_date_ms": "1494580145000",
			"expires_date": "2017-05-12 09:09:05 Etc/GMT",
			"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
			"original_purchase_date_ms": "1491988147000",
			"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
			"purchase_date_pst": "2017-04-12 02:09:05 America/Los_Angeles",
			"purchase_date_ms": "1491988145000",
			"purchase_date": "2017-04-12 09:09:05 Etc/GMT",
			"original_transaction_id": "520000222666210",
			"transaction_id": "520000222666210",
			"product_id": "test.product.2",
			"quantity": "1"
		}, {
			"is_trial_period": "false",
			"web_order_line_item_id": "520000040310204",
			"expires_date_pst": "2017-06-12 02:09:05 America/Los_Angeles",
			"expires_date_ms": "1497258545000",
			"expires_date": "2017-06-12 09:09:05 Etc/GMT",
			"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
			"original_purchase_date_ms": "1491988147000",
			"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
			"purchase_date_pst": "2017-05-12 02:09:05 America/Los_Angeles",
			"purchase_date_ms": "1494580145000",
			"purchase_date": "2017-05-12 09:09:05 Etc/GMT",
			"original_transaction_id": "520000222666210",
			"transaction_id": "520000229967375",
			"product_id": "test.product.2",
			"quantity": "1"
		}, {
			"is_trial_period": "false",
			"web_order_line_item_id": "520000042429692",
			"expires_date_pst": "2017-07-12 02:09:05 America/Los_Angeles",
			"expires_date_ms": "1499850545000",
			"expires_date": "2017-07-12 09:09:05 Etc/GMT",
			"original_purchase_date_pst": "2017-04-12 02:09:07 America/Los_Angeles",
			"original_purchase_date_ms": "1491988147000",
			"original_purchase_date": "2017-04-12 09:09:07 Etc/GMT",
			"purchase_date_pst": "2017-06-12 02:09:05 America/Los_Angeles",
			"purchase_date_ms": "1497258545000",
			"purchase_date": "2017-06-12 09:09:05 Etc/GMT",
			"original_transaction_id": "520000222666210",
			"transaction_id": "520000237814537",
			"product_id": "test.product.2",
			"quantity": "1"
		}],
		"original_application_version": "REMOVED FOR PRIVACY",
		"original_purchase_date_pst": "2017-04-11 03:08:56 America/Los_Angeles",
		"original_purchase_date_ms": "1491905336000",
		"original_purchase_date": "2017-04-11 10:08:56 Etc/GMT",
		"request_date_pst": "2017-07-04 00:20:22 America/Los_Angeles",
		"request_date_ms": "1499152822045",
		"request_date": "2017-07-04 07:20:22 Etc/GMT",
		"receipt_creation_date_pst": "2017-06-30 00:28:13 America/Los_Angeles",
		"receipt_creation_date_ms": "1498807693000",
		"receipt_creation_date": "2017-06-30 07:28:13 Etc/GMT",
		"version_external_identifier": 0,
		"download_id": 92022481841993,
		"application_version": "REMOVED FOR PRIVACY",
		"bundle_id": "REMOVED FOR PRIVACY",
		"app_item_id": 0,
		"adam_id": 0,
		"receipt_type": "Production"
	},
	"environment": "Production",
	"status": 0
}
```

Test case

```js
/**
 * Testing apple receipt
 */

const iap     = require( "in-app-purchase" ),
      receipt = require( "./test.receipt.json" );

// console.log( iap.getPurchaseData( receipt ).length ); // prints 6
console.log( iap.getPurchaseData( receipt ) );
```

Output 

```
[ { isTrialPeriod: 'false',
    originalPurchaseDatePst: '2017-04-11 03:10:23 America/Los_Angeles',
    originalPurchaseDateMs: 1491905423000,
    originalPurchaseDate: '2017-04-11 10:10:23 Etc/GMT',
    purchaseDatePst: '2017-04-11 03:10:23 America/Los_Angeles',
    purchaseDateMs: 1491905423000,
    purchaseDate: '2017-04-11 10:10:23 Etc/GMT',
    originalTransactionId: 520000222450197,
    transactionId: '520000222450197',
    productId: 'db.test.product.1',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 0 },
  { isTrialPeriod: 'false',
    webOrderLineItemId: 520000040310204,
    expiresDatePst: '2017-06-12 02:09:05 America/Los_Angeles',
    expiresDateMs: 1497258545000,
    expiresDate: '2017-06-12 09:09:05 Etc/GMT',
    originalPurchaseDatePst: '2017-04-12 02:09:07 America/Los_Angeles',
    originalPurchaseDateMs: 1491988147000,
    originalPurchaseDate: '2017-04-12 09:09:07 Etc/GMT',
    purchaseDatePst: '2017-05-12 02:09:05 America/Los_Angeles',
    purchaseDateMs: 1494580145000,
    purchaseDate: '2017-05-12 09:09:05 Etc/GMT',
    originalTransactionId: 520000222666210,
    transactionId: '520000229967375',
    productId: 'test.product.2',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 1497258545000 },
  { isTrialPeriod: 'false',
    webOrderLineItemId: 520000042429692,
    expiresDatePst: '2017-07-12 02:09:05 America/Los_Angeles',
    expiresDateMs: 1499850545000,
    expiresDate: '2017-07-12 09:09:05 Etc/GMT',
    originalPurchaseDatePst: '2017-04-12 02:09:07 America/Los_Angeles',
    originalPurchaseDateMs: 1491988147000,
    originalPurchaseDate: '2017-04-12 09:09:07 Etc/GMT',
    purchaseDatePst: '2017-06-12 02:09:05 America/Los_Angeles',
    purchaseDateMs: 1497258545000,
    purchaseDate: '2017-06-12 09:09:05 Etc/GMT',
    originalTransactionId: 520000222666210,
    transactionId: '520000237814537',
    productId: 'test.product.2',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 1499850545000 },
  { isTrialPeriod: 'false',
    originalPurchaseDatePst: '2017-04-11 03:10:23 America/Los_Angeles',
    originalPurchaseDateMs: 1491905423000,
    originalPurchaseDate: '2017-04-11 10:10:23 Etc/GMT',
    purchaseDatePst: '2017-04-11 03:10:23 America/Los_Angeles',
    purchaseDateMs: 1491905423000,
    purchaseDate: '2017-04-11 10:10:23 Etc/GMT',
    originalTransactionId: 520000222450197,
    transactionId: '520000222450197',
    productId: 'de.test.product.1',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 0 },
  { isTrialPeriod: 'false',
    webOrderLineItemId: 520000040310204,
    expiresDatePst: '2017-06-12 02:09:05 America/Los_Angeles',
    expiresDateMs: 1497258545000,
    expiresDate: '2017-06-12 09:09:05 Etc/GMT',
    originalPurchaseDatePst: '2017-04-12 02:09:07 America/Los_Angeles',
    originalPurchaseDateMs: 1491988147000,
    originalPurchaseDate: '2017-04-12 09:09:07 Etc/GMT',
    purchaseDatePst: '2017-05-12 02:09:05 America/Los_Angeles',
    purchaseDateMs: 1494580145000,
    purchaseDate: '2017-05-12 09:09:05 Etc/GMT',
    originalTransactionId: 520000222666210,
    transactionId: '520000229967375',
    productId: 'db.test.product.2',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 1497258545000 },
  { isTrialPeriod: 'false',
    webOrderLineItemId: 520000042429692,
    expiresDatePst: '2017-07-12 02:09:05 America/Los_Angeles',
    expiresDateMs: 1499850545000,
    expiresDate: '2017-07-12 09:09:05 Etc/GMT',
    originalPurchaseDatePst: '2017-04-12 02:09:07 America/Los_Angeles',
    originalPurchaseDateMs: 1491988147000,
    originalPurchaseDate: '2017-04-12 09:09:07 Etc/GMT',
    purchaseDatePst: '2017-06-12 02:09:05 America/Los_Angeles',
    purchaseDateMs: 1497258545000,
    purchaseDate: '2017-06-12 09:09:05 Etc/GMT',
    originalTransactionId: 520000222666210,
    transactionId: '520000237814537',
    productId: 'db.test.product.2',
    quantity: 1,
    bundleId: 'REMOVED FOR PRIVACY',
    expirationDate: 1499850545000 } ]
```

As you can see I have 4 purchases ( they are the same in `in_app` and `latest_receipt_info` ), but have 6 overall in the response of `getPurchaseData`.

**Comment by voltrue2** [comment]
Hello

Thank you so much for this info. It looks like we do have a duplicate bug here.
I will look into this as soon as possible.

Cheers!

**Comment by voltrue2** [comment]
Hello
The problem is caused by `purchase_date_ms` of multiple items have the same value.
So the block of code to prevent duplicate is not catching the items with same `original_transaction_id`
and that is causing the duplication.

**Comment by voltrue2** [comment]
I have changed the duplicate check to evaludate as following in the develop branch: 133e1fb8cea1c78f834ba4639b17b51c3c411c08

This should at least allow us to deal with items with the same timestamp...

**Comment by drinchev** [comment]
The thing is that I always receive the same purchases into `in_app` filed of the receipt and in the `latest_receipt_info`. I'm wondering if this is the general behaviour of Apple ( Our app is using renewing purchases ).

Thanks for your time!

**Comment by voltrue2** [comment]
According to Apple: https://developer.apple.com/library/content/releasenotes/General/ValidateAppStoreReceipt/Chapters/ValidateRemotely.html

`latest_receipt_info` is returned only with iOS 6 style receipts for auto-renewable subscriptions. I wonder if your client-side has updated the purchase transaction part of the code?

As far as I am aware, we should only be checking `in_app` filed as it is the current format.

Cheers

**Comment by drinchev** [comment]
Yeah there's a whole fire on apple forums about this [1], [2], [3].

tl;dr : It's some sort of misleading docs.

1 : https://forums.developer.apple.com/thread/69531
2 : https://forums.developer.apple.com/message/142334#142334
3 : https://forums.developer.apple.com/thread/69531#205328

To my knowledge and experience sometimes the `in_app` and `latest_receipt_info` contains the same transactions, sometimes not.

**Comment by voltrue2** [comment]
Closing unless we run into more issue with this. :)

**Comment by jbgomez21** [comment]
Hi @voltrue2 

I'm using 1.8.4 version and getPurchaseDate still return duplicates, In the next list there are 12 elements, 6 purchase and 6 duplicates.

I'm interpreting something wrong?

Regards!


```
[
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343576442",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:34:52 Etc/GMT",
        "purchaseDateMs": 1507995292000,
        "purchaseDatePst": "2017-10-14 08:34:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:37:52 Etc/GMT",
        "expiresDateMs": 1507995472000,
        "expiresDatePst": "2017-10-14 08:37:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570731,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507995472000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343577314",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:40:52 Etc/GMT",
        "purchaseDateMs": 1507995652000,
        "purchaseDatePst": "2017-10-14 08:40:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:43:52 Etc/GMT",
        "expiresDateMs": 1507995832000,
        "expiresDatePst": "2017-10-14 08:43:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570740,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507995832000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343578195",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:46:52 Etc/GMT",
        "purchaseDateMs": 1507996012000,
        "purchaseDatePst": "2017-10-14 08:46:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:49:52 Etc/GMT",
        "expiresDateMs": 1507996192000,
        "expiresDatePst": "2017-10-14 08:49:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570760,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507996192000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344359860",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 02:59:56 Etc/GMT",
        "purchaseDateMs": 1508295596000,
        "purchaseDatePst": "2017-10-17 19:59:56 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:02:56 Etc/GMT",
        "expiresDateMs": 1508295776000,
        "expiresDatePst": "2017-10-17 20:02:56 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601333,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508295776000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344361490",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 03:11:19 Etc/GMT",
        "purchaseDateMs": 1508296279000,
        "purchaseDatePst": "2017-10-17 20:11:19 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:14:19 Etc/GMT",
        "expiresDateMs": 1508296459000,
        "expiresDatePst": "2017-10-17 20:14:19 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601417,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508296459000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344363075",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 03:17:19 Etc/GMT",
        "purchaseDateMs": 1508296639000,
        "purchaseDatePst": "2017-10-17 20:17:19 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:20:19 Etc/GMT",
        "expiresDateMs": 1508296819000,
        "expiresDatePst": "2017-10-17 20:20:19 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601451,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508296819000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343576442",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:34:52 Etc/GMT",
        "purchaseDateMs": 11507995292000,
        "purchaseDatePst": "2017-10-14 08:34:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:37:52 Etc/GMT",
        "expiresDateMs": 1507995472000,
        "expiresDatePst": "2017-10-14 08:37:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570731,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507995472000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343577314",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:40:52 Etc/GMT",
        "purchaseDateMs": 1507995652000,
        "purchaseDatePst": "2017-10-14 08:40:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:43:52 Etc/GMT",
        "expiresDateMs": 1507995832000,
        "expiresDatePst": "2017-10-14 08:43:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570740,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507995832000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000343578195",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-14 15:46:52 Etc/GMT",
        "purchaseDateMs": 1507996012000,
        "purchaseDatePst": "2017-10-14 08:46:52 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-14 15:49:52 Etc/GMT",
        "expiresDateMs": 1507996192000,
        "expiresDatePst": "2017-10-14 08:49:52 America/Los_Angeles",
        "webOrderLineItemId": 1000000036570760,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1507996192000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344359860",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 02:59:56 Etc/GMT",
        "purchaseDateMs": 1508295596000,
        "purchaseDatePst": "2017-10-17 19:59:56 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:02:56 Etc/GMT",
        "expiresDateMs": 1508295776000,
        "expiresDatePst": "2017-10-17 20:02:56 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601333,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508295776000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344361490",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 03:11:19 Etc/GMT",
        "purchaseDateMs": 1508296279000,
        "purchaseDatePst": "2017-10-17 20:11:19 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:14:19 Etc/GMT",
        "expiresDateMs": 1508296459000,
        "expiresDatePst": "2017-10-17 20:14:19 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601417,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508296459000
    },
    {
        "quantity": 1,
        "productId": "com.nostressapps.colorart.weekly",
        "transactionId": "1000000344363075",
        "originalTransactionId": "1000000343576417",
        "purchaseDate": "2017-10-18 03:17:19 Etc/GMT",
        "purchaseDateMs": 1508296639000,
        "purchaseDatePst": "2017-10-17 20:17:19 America/Los_Angeles",
        "originalPurchaseDate": "2017-10-14 15:31:55 Etc/GMT",
        "originalPurchaseDateMs": 1507995115000,
        "originalPurchaseDatePst": "2017-10-14 08:31:55 America/Los_Angeles",
        "expiresDate": "2017-10-18 03:20:19 Etc/GMT",
        "expiresDateMs": 1508296819000,
        "expiresDatePst": "2017-10-17 20:20:19 America/Los_Angeles",
        "webOrderLineItemId": 1000000036601451,
        "isTrialPeriod": "false",
        "bundleId": "com.nostressapps.colorart",
        "expirationDate": 1508296819000
    }
]
```
